jarbas/core/management/commands/companies.py

```diff
@@ -18,50 +22,56 @@ def handle(self, *args, **options):
 
         if options.get('drop', False):
             self.drop_all(Company)
-            self.drop_all(Activity)
+            #self.drop_all(Activity)
             self.count = 0
 
-        self.save_companies()
-
+        self.bulk_create_by(self.companies, options['batch_size'])
+    
+    
     def save_companies(self):
+        self.bulk_create_by(self.companies, 10000)
+
+    @property
+    def companies(self):
         """
         Receives path to the dataset file and create a Company object for
         each row of each file. It creates the related activity when needed.
         """
-        skip = ('main_activity', 'secondary_activity')
+        skip=('main_acvtivity', 'secondary_activity')
         keys = list(f.name for f in Company._meta.fields if f not in skip)
         with lzma.open(self.path, mode='rt') as file_handler:
-            for row in csv.DictReader(file_handler):
+            for row in csv.DictReader(file_handler): 
                 main, secondary = self.save_activities(row)
-
+                
                 filtered = {k: v for k, v in row.items() if k in keys}
-                obj = Company.objects.create(**self.serialize(filtered))
-                for activity in main:
-                    obj.main_activity.add(activity)
-                for activity in secondary:
-                    obj.secondary_activity.add(activity)
-                obj.save()
-
-                self.count += 1
-                self.print_count(Company, count=self.count)
+                filtered['main_activity'] = [main]
+                
+                if len(secondary) == 0:
+                    filtered['secondary_activity'] = []
+                else:
+                    filtered['secondary_activity'] = [secondary]
+                obj = Company(**self.serialize(filtered))
+                yield obj
+                 
+                #obj.save()
+                #
+                #self.count += 1
+                #self.print_count(Company, count=self.count)
 
     def save_activities(self, row):
```

Comment:
If `Activity` is not a model anymore, I'd add the logic to _parse_ the data from teh CSV into a dict to the `serialize` method, not here… this method is called _save_ something, but it doesn't save anything, but actually serialize activities ; )